server/routes/admin/html.py

```diff
@@ -72,11 +98,13 @@ def load_data():
   command2 = [
       'python',
       '-m',
-      'tools.nl.embeddings.build_custom_dc_embeddings',
-      '--input_dir',
-      f'{nl_dir}',
+      'tools.nl.embeddings.build_embeddings',
+      '--embeddings_name',
+      _CUSTOM_EMBEDDING_INDEX,
       '--output_dir',
       f'{nl_embeddings_dir}',
+      '--catalog',
+      json.dumps(custom_catalog_dict),
```

Comment:
Do we want the offline script to also specify the catalog dict? Would it be simpler to pass just the `source_path` and `embeddings_path` and have the embeddings tool create the full dict internally?

Reply:
In a regular flow (like adding base DC embedding index, for example, undata), we would:

1. create the entry in catalog.yaml manually
2. run build_embeddings tool to create embedding csv (w/o pre-defined output path).
3. use the catalog.yaml to bring up nl server.

The process here actually mimics that flow.


In the old way of producing / using custom_catalog.yaml, the yaml was not really used as a config (which should be input), but rather an output.. that makes the build embeddings tool not generalizable (an effect of deviating the natural process mentioned above). That also makes the embeddings_path and embeddings source csv path spread across nl server, website(admin) and build_embeddings tool. Now this file would be the source of truth and provide arguments for all the rest.

Also after migration to the script, we should have a step of creating this catalog.yaml as a pre-step before running all the stages below as well.

Reply:
OK. Let's get this PR and @ajaits's PR in and we can work on the script changes following that.